vt-digital-libraries-platform/dlp-ingest
# Refactor metadata ingest wholesale

**Refactor metadata ingest.**
* * *

**asana Ticket**: (https://app.asana.com/0/1203932369414348/1206697305609590) (:star:)

* Other Relevant Links (Meeting note, project page, related pull requests, etc.)

# What does this Pull Request do? (:star:)
This is a pretty big refactor of metadata ingest. It removes support for ingesting legacy fields, removes some conditionals that no longer necessary (were they ever?) and tries to consolidate some env vars and make them more descriptive. I think this could actually still use some thought. But I think this is fine. ...for now.

# What's the changes? (:star:)
* Errthing lol
* We kept a lot of the helper functions and the basic algorithm is the same as [S3toDDB](https://github.com/vt-digital-libraries-platform/S3toDDB) but it was a pretty heavy refactor.

# How should this be tested?
* I think you should test this by deleting the `nmcst` Archive records that are probably ALL crashing pages. I can help with deleting these if y'all need me to. It can be done pretty easily in the Dynamo console. Don't delete everything, just `nmcst` though
* Then I would use this branch `wlh_20240117_refactor` to re-ingest the `nmcst` records. You might need to adjust your [calling script](https://github.com/vt-digital-libraries-platform/dlp-ingest/blob/main/lambda_local.sh) to the newer variable names
* Then check to see if the issues with `rights_holder` and `display_date` are resolved. They should be ingested into the Dynamo Archive records as Arrays instead of Strings. ...And also they should stop crashing their pages. Lol

# Additional Notes:
Any additional information that you think would be helpful when reviewing this PR.
* branch: `wlh_20240117_refactor`
* I'm including Kai too because reviewing the lambda_function.py changes is gonna be tough. It's a lot to look through.

# Interested parties
Tag (@ mention) interested parties
@goynejennifer 
@otokama 

(:star:) Required fields
